scripts/adlsgen2setup.py

```diff
@@ -0,0 +1,177 @@
+import argparse
+import asyncio
+import json
+import logging
+import os
+
+import aiohttp
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.identity.aio import AzureDeveloperCliCredential
+from azure.storage.filedatalake.aio import (
+    DataLakeDirectoryClient,
+    DataLakeServiceClient,
+)
+
+
+class AdlsGen2Setup:
+    """
+    Sets up a Data Lake Storage Gen 2 account with sample data and access control
+    """
+
+    def __init__(
+        self,
+        data_directory: str,
+        storage_account_name: str,
+        filesystem_name: str,
+        security_enabled_groups: bool,
+        data_access_control_format: dict[str, any],
+        credentials: AsyncTokenCredential,
+    ):
+        """
+        Initializes the command
+
+        Parameters
+        ----------
+        data_directory
+            Directory where sample files are located
+        storage_account_name
+            Name of the Data Lake Storage Gen 2 account to use
+        filesystem_name
+            Name of the container / filesystem in the Data Lake Storage Gen 2 account to use
+        security_enabled_groups
+            When creating groups in Azure AD, whether or not to make them security enabled
+        data_access_control_format
+            File describing how to create groups, upload files with access control. See the sampleacls.json for the format of this file
+        """
+        self.data_directory = data_directory
+        self.storage_account_name = storage_account_name
+        self.filesystem_name = filesystem_name
+        self.credentials = credentials
+        self.security_enabled_groups = security_enabled_groups
+        self.data_access_control_format = data_access_control_format
+
+    async def run(self):
+        async with DataLakeServiceClient(
```

Comment:
I notice you use "async with" for both the client and credentials, and I think in other scripts we just create the clients and pass them around. I'm curious if the "async with" context manager approach is beneficial in some way? Should we be using it elsewhere?

Reply:
if you pass the normal clients around they make blocking http calls. Using async takes advantage of aiohttp non blocking behavior.

Reply:
Huh, even if you pass around a client from the aio module? I think that means our current code isn't set up well. Was there documentation around that? (I believe you, just trying to understand better myself)

Reply:
so the problem is you'll randomly see "client session unclosed" errors. A cheap way to reproduce this is to force close the app when it's serving a streaming chat session


Here's the GH PR / has some linked issues: https://github.com/Azure/azure-sdk-for-python/pull/9090